### ISSUE TITLE
mod  breaks step assist on other mods

### ISSUE THREAD

**Issue by PokerfaceCactus** [Alex-the-666/Ice_and_Fire#633]
For some reason adding Ice and Fire to my mods just breaks every single method of getting step assist from other mods

known mods that get effected by this:
Mekanism (free runners)
Everlasting abilities
Draconic evolution

there's a few more but i can't remember the name from the top of my head.

basicly step assist in every single mod we tested gets broken when adding this mod and it's really annoying if you're used to having it almost always.

**Comment by laserlemons** [comment]
Horses also have no more step assist.

**Comment by AlexModGuy** [comment]
what is step assist

**Comment by PlatinumFox** [comment]
@Alex-the-666 I think they mean that horses (and apparently stuff from other mods) no longer auto jump 1 block for smooth uphill movement

**Comment by laserlemons** [comment]
This is the same probelm as #567

**Comment by Ormei** [comment]
Step assist in this context has nothing to do with jumping.
**Jumping**: The player can jump 1 block, some horses about 5 block. Jumping beacon increases this.
**Step assist**: The player can step up 0,5 block (slabs/stairs) and vanilla horses 1 block (they just walk up, no jumping, no auto jumping assist). Step assist grants the ability for players to step up 1 block like horses do.

With this mod it is broken, any horse or any player with increased step height can't just walk up more than 0,5 block (slabs/stairs).

PS: Tested with Quality Tools mod - also broken.

**Comment by Holgath** [comment]
I am having the same problem.  Step assist doesn't work at all in any of the mods I have tried with.  When I downgrade only Ice and Fire to the previous version they all start working again.   I don't know much about coding but it looks like it is called stepHeight in the code.  Hopefully this information helps.

**Comment by Leyal-Astlcia** [comment]
event\EventLiving.java (207)
`event.getEntityLiving().stepHeight = 0.6F;`

This row cause this issue.
This issue will be solved temporary if delete this row.
However, this is not correct behavior.
(It seems works correctly, but will probably cause another issue after encountering Seiren etc)

I wait for right correction by creator.

**Comment by laneyfamily** [comment]
Can confirm this behavior with [StepUp ](https://minecraft.curseforge.com/projects/stepup?gameCategorySlug=mc-mods&projectID=239236#c69) as well

**Comment by NotTooManyItems** [comment]
I have released a version that should work with step assist mods. Check it out and let me know how it goes for you.

https://github.com/NotTooManyItems/Ice_and_Fire/releases/tag/1.4.0

**Comment by SpringsTS** [comment]
@NotTooManyItems, I tested your version several minutes, I confirm it fixes the problem. Thank you!
I will play more and check if everything else works.

**Comment by Ormei** [comment]
Briefly tested the new version,
*Quality tools* and *vanilla horses* step up works, thanks!
But now this version is incompatible with Thaumcraft6. Does it have something to do with Thaumcraft API inside this repo?

<details><summary>crashlog</summary>
<p>

```
---- Minecraft Crash Report ----

WARNING: coremods are present:
  MalisisSwitchesPlugin (malisisswitches-1.12.2-5.1.0.jar)
  MalisisCorePlugin (malisiscore-1.12.2-6.4.0.jar)
  ItemPatchingLoader (ItemPhysic_Full+1.4.11_mc1.12.2.jar)
  ShoulderPlugin (ShoulderSurfing-1.12.2-1.12.jar)
  IELoadingPlugin (ImmersiveEngineering-core-0.12-82.jar)
  BCModPlugin (backpacks+1.12.2+-+3.4.4.jar)
  llibrary (llibrary-1.7.9-1.12.2.jar)
  NWRTweak (redstonepaste-mc1.12-1.7.5.jar)
  BetterFoliageLoader (BetterFoliage-MC1.12-2.1.10.jar)
  CreativePatchingLoader (CreativeCore_v1.9.22_mc1.12.2.jar)
  SSLoadingPlugin (SereneSeasons-1.12.2-1.0.0-universal.jar)
Contact their authors BEFORE contacting forge

// Uh... Did I do that?

Time: 5/14/18 8:42 PM
Description: There was a severe problem during mod loading that has caused the game to fail

net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from Thaumcraft (thaumcraft)
Caused by: java.lang.NoSuchMethodError: thaumcraft.api.internal.CommonInternals.generateUniqueItemstackId(Lnet/minecraft/item/ItemStack;)I
	at thaumcraft.common.lib.crafting.ThaumcraftCraftingManager.getObjectTags(ThaumcraftCraftingManager.java:147)
	at thaumcraft.common.lib.crafting.ThaumcraftCraftingManager.getObjectTags(ThaumcraftCraftingManager.java:140)
	at thaumcraft.common.lib.InternalMethodHandler.getObjectAspects(InternalMethodHandler.java:86)
	at thaumcraft.api.aspects.AspectHelper.getObjectAspects(AspectHelper.java:69)
	at thaumcraft.api.aspects.AspectList.<init>(AspectList.java:21)
	at thaumcraft.common.config.ConfigAspects.registerItemAspects(ConfigAspects.java:157)
	at thaumcraft.common.config.ConfigAspects.postInit(ConfigAspects.java:25)
	at thaumcraft.proxies.CommonProxy.postInit(CommonProxy.java:82)
	at thaumcraft.proxies.ClientProxy.postInit(ClientProxy.java:53)
	at thaumcraft.Thaumcraft.postInit(Thaumcraft.java:53)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:600)
	at sun.reflect.GeneratedMethodAccessor3.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:280)
	at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:258)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
	at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
	at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
	at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
	at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
	at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
	at com.google.common.eventbus.EventBus.post(EventBus.java:217)
	at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:149)
	at net.minecraftforge.fml.common.Loader.initializeMods(Loader.java:760)
	at net.minecraftforge.fml.client.FMLClientHandler.finishMinecraftLoading(FMLClientHandler.java:352)
	at net.minecraft.client.Minecraft.func_71384_a(Minecraft.java:534)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:377)
	at net.minecraft.client.main.Main.main(SourceFile:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- System Details --
Details:
	Minecraft Version: 1.12.2
	Operating System: Windows 7 (amd64) version 6.1
	Java Version: 1.8.0_91, Oracle Corporation
	Java VM Version: Java HotSpot(TM) 64-Bit Server VM (mixed mode), Oracle Corporation
	Memory: 840190208 bytes (801 MB) / 1998585856 bytes (1906 MB) up to 7462518784 bytes (7116 MB)
	JVM Flags: 6 total; -Xmx7G -XX:+UseConcMarkSweepGC -XX:+CMSIncrementalMode -XX:-UseAdaptiveSizePolicy -Xmn512M -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump
	IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
	FML: MCP 9.42 Powered by Forge 14.23.3.2681 Optifine OptiFine_1.12.2_HD_U_D2 66 mods loaded, 66 mods active
	States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored

	| State | ID                        | Version          | Source                                        | Signature                                |
	|:----- |:------------------------- |:---------------- |:--------------------------------------------- |:---------------------------------------- |
	| UCHIJ | minecraft                 | 1.12.2           | minecraft.jar                                 | None                                     |
	| UCHIJ | mcp                       | 9.42             | minecraft.jar                                 | None                                     |
	| UCHIJ | FML                       | 8.0.99.99        | forge-1.12.2-14.23.3.2681.jar                 | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJ | forge                     | 14.23.3.2681     | forge-1.12.2-14.23.3.2681.jar                 | e3c3d50c7c986df74c645c0ac54639741c90a557 |
	| UCHIJ | creativecoredummy         | 1.0.0            | minecraft.jar                                 | None                                     |
	| UCHIJ | itemphysic                | 1.4.0            | minecraft.jar                                 | None                                     |
	| UCHIJ | antiqueatlas              | 4.4.7            | antiqueatlas-1.12.2-4.4.7.jar                 | None                                     |
	| UCHIJ | antiqueatlasoverlay       | 1.2              | antiqueatlas-1.12.2-4.4.7.jar                 | None                                     |
	| UCHIJ | examplemod                | 1.0              | antiqueatlas-1.12.2-4.4.7.jar                 | None                                     |
	| UCHIJ | jei                       | 4.8.5.159        | jei_1.12.2-4.8.5.159.jar                      | None                                     |
	| UCHIJ | appleskin                 | 1.0.9            | AppleSkin-mc1.12-1.0.9.jar                    | None                                     |
	| UCHIJ | astikoor                  | 1.0.0            | astikoor-1.0.0.jar                            | None                                     |
	| UCHIJ | backpacks16840            | 3.4.4            | backpacks+1.12.2+-+3.4.4.jar                  | None                                     |
	| UCHIJ | baubles                   | 1.5.2            | Baubles-1.12-1.5.2.jar                        | None                                     |
	| UCHIJ | betterfoliage             | 2.1.10           | BetterFoliage-MC1.12-2.1.10.jar               | None                                     |
	| UCHIJ | bibliocraft               | 2.4.3            | BiblioCraft[v2.4.3][MC1.12.0].jar             | None                                     |
	| UCHIJ | biomesoplenty             | 7.0.1.2325       | BiomesOPlenty-1.12.2-7.0.1.2325-universal.jar | None                                     |
	| UCHIJ | borntobe                  | @VERSION@        | borntobe-1.12.2-1.0.jar                       | None                                     |
	| UCHIJ | codechickenlib            | 3.1.8.341        | CodeChickenLib-1.12.2-3.1.8.341-universal.jar | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
	| UCHIJ | crafttweaker              | 4.1.6            | CraftTweaker2-1.12-4.1.6.jar                  | None                                     |
	| UCHIJ | ctgui                     | 1.0.0            | CraftTweaker2-1.12-4.1.6.jar                  | None                                     |
	| UCHIJ | crafttweakerjei           | 2.0.0            | CraftTweaker2-1.12-4.1.6.jar                  | None                                     |
	| UCHIJ | creativecore              | 1.9.9            | CreativeCore_v1.9.22_mc1.12.2.jar             | None                                     |
	| UCHIJ | customspawner             | 3.11.4           | CustomMobSpawner-3.11.4.jar                   | None                                     |
	| UCHIJ | mocreatures               | 12.0.4           | DrZharks MoCreatures Mod-12.0.4.jar           | None                                     |
	| UCHIJ | embers                    | 0.230            | embers-0.230.jar                              | None                                     |
	| UCHIJ | enderstorage              | 2.4.2.126        | EnderStorage-1.12.2-2.4.2.126-universal.jar   | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
	| UCHIJ | enderzoo                  | 1.12.1-1.4.0.49  | EnderZoo-1.12.1-1.4.0.49.jar                  | None                                     |
	| UCHIJ | forgemultipartcbe         | 2.4.2.58         | ForgeMultipart-1.12.2-2.4.2.58-universal.jar  | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
	| UCHIJ | microblockcbe             | 2.4.2.58         | ForgeMultipart-1.12.2-2.4.2.58-universal.jar  | None                                     |
	| UCHIJ | minecraftmultipartcbe     | 2.4.2.58         | ForgeMultipart-1.12.2-2.4.2.58-universal.jar  | None                                     |
	| UCHIJ | gravestone                | 1.0.14           | GraveStone-1.12-Graves-1.0.14.jar             | None                                     |
	| UCHIJ | gravestone-extended       | 1.2.6            | GraveStone-1.12.2-Extended-1.2.6.jar          | None                                     |
	| UCHIJ | hopperducts               | 1.5              | hopperducts-mc1.12-1.5.jar                    | None                                     |
	| UCHIJ | llibrary                  | 1.7.9            | llibrary-1.7.9-1.12.2.jar                     | None                                     |
	| UCHIJ | iceandfire                | 1.4.0            | iceandfire-1.4.0(unofficial).jar              | None                                     |
	| UCHIJ | immersiveengineering      | 0.12-82          | ImmersiveEngineering-0.12-82.jar              | 4cb49fcde3b43048c9889e0a3d083225da926334 |
	| UCHIJ | immersivepetroleum        | 1.1.9            | immersivepetroleum-1.12.2-1.1.9.jar           | None                                     |
	| UCHIJ | jepb                      | 1.2.1            | jepb-1.12-1.2.1.jar                           | None                                     |
	| UCHIJ | justenoughdimensions      | 1.5.5            | justenoughdimensions-1.12.1-1.5.5.jar         | 2b03e1423915a189b8094816baa18f239d576dff |
	| UCHIJ | malisiscore               | 1.12.2-6.4.0     | malisiscore-1.12.2-6.4.0.jar                  | None                                     |
	| UCHIJ | malisisblocks             | 1.12.2-6.1.0     | malisisblocks-1.12.2-6.1.0.jar                | None                                     |
	| UCHIJ | malisisdoors              | 1.12.2-7.3.0     | malisisdoors-1.12.2-7.3.0.jar                 | None                                     |
	| UCHIJ | malisisswitches           | 1.12.2-5.1.0     | malisisswitches-1.12.2-5.1.0.jar              | None                                     |
	| UCHIJ | mapwriter                 | 2.8.2            | MapWriter-1.12.2-2.8.2.jar                    | None                                     |
	| UCHIJ | mrtjpcore                 | 2.1.1.27         | MrTJPCore-1.12.2-2.1.1.27-universal.jar       | None                                     |
	| UCHIJ | nbtedit                   | $version         | NBTEdit-0.7.jar                               | None                                     |
	| UCHIJ | recipehandler             | 0.10             | NoMoreRecipeConflict-0.10(1.12.2).jar         | None                                     |
	| UCHIJ | potioncore                | 1.6_for_1.12.2   | PotionCore-1.6_for_1.12.2.jar                 | None                                     |
	| UCHIJ | projectred-core           | 4.8.6.74         | ProjectRed-1.12.2-4.8.6.74-Base.jar           | None                                     |
	| UCHIJ | projectred-integration    | 4.8.6.74         | ProjectRed-1.12.2-4.8.6.74-integration.jar    | None                                     |
	| UCHIJ | projectred-transmission   | 4.8.6.74         | ProjectRed-1.12.2-4.8.6.74-integration.jar    | None                                     |
	| UCHIJ | projectred-fabrication    | 4.8.6.74         | ProjectRed-1.12.2-4.8.6.74-fabrication.jar    | None                                     |
	| UCHIJ | projectred-illumination   | 4.8.6.74         | ProjectRed-1.12.2-4.8.6.74-lighting.jar       | None                                     |
	| UCHIJ | projectred-expansion      | 4.8.6.74         | ProjectRed-1.12.2-4.8.6.74-mechanical.jar     | None                                     |
	| UCHIJ | projectred-transportation | 4.8.6.74         | ProjectRed-1.12.2-4.8.6.74-mechanical.jar     | None                                     |
	| UCHIJ | projectred-exploration    | 4.8.6.74         | ProjectRed-1.12.2-4.8.6.74-world.jar          | None                                     |
	| UCHIJ | qualitytools              | 1.0.6_for_1.12.2 | QualityTools-1.0.6_for_1.12.2.jar             | None                                     |
	| UCHIJ | redstonepaste             | 1.7.5            | redstonepaste-mc1.12-1.7.5.jar                | None                                     |
	| UCHIJ | roguelike                 | 1.8.0            | RoguelikeDungeons-1.12.2-1.8.0.jar            | None                                     |
	| UCHIJ | sereneseasons             | 1.0.0            | SereneSeasons-1.12.2-1.0.0-universal.jar      | None                                     |
	| UCHIJ | shouldersurfing           | 1.12             | ShoulderSurfing-1.12.2-1.12.jar               | d6261bb645f41db84c74f98e512c2bb43f188af2 |
	| UCHIJ | simpleportals             | 1.12.2-1.5.1.0   | simpleportals-1.12.2-1.5.1.0.jar              | None                                     |
	| UCHIE | thaumcraft                | 6.1.BETA13       | Thaumcraft-1.12.2-6.1.BETA13.jar              | None                                     |
	| UCHIJ | twilightforest            | 3.6.345          | twilightforest-1.12.2-3.6.345-universal.jar   | None                                     |
	| UCHIJ | weaponmod                 | v1.19.2.5        | weaponmod-1.19.2.5.jar                        | None                                     |

	Loaded coremods (and transformers): 
MalisisSwitchesPlugin (malisisswitches-1.12.2-5.1.0.jar)
  
MalisisCorePlugin (malisiscore-1.12.2-6.4.0.jar)
  
ItemPatchingLoader (ItemPhysic_Full+1.4.11_mc1.12.2.jar)
  com.creativemd.itemphysic.ItemTransformer
ShoulderPlugin (ShoulderSurfing-1.12.2-1.12.jar)
  com.teamderpy.shouldersurfing.asm.ShoulderTransformations
IELoadingPlugin (ImmersiveEngineering-core-0.12-82.jar)
  blusunrize.immersiveengineering.common.asm.IEClassTransformer
BCModPlugin (backpacks+1.12.2+-+3.4.4.jar)
  brad16840.core.ClassTransformer
llibrary (llibrary-1.7.9-1.12.2.jar)
  net.ilexiconn.llibrary.server.core.plugin.LLibraryTransformer
  net.ilexiconn.llibrary.server.core.patcher.LLibraryRuntimePatcher
NWRTweak (redstonepaste-mc1.12-1.7.5.jar)
  net.fybertech.nwr.NWRTransformer
BetterFoliageLoader (BetterFoliage-MC1.12-2.1.10.jar)
  mods.betterfoliage.loader.BetterFoliageTransformer
CreativePatchingLoader (CreativeCore_v1.9.22_mc1.12.2.jar)
  
SSLoadingPlugin (SereneSeasons-1.12.2-1.0.0-universal.jar)
  sereneseasons.asm.transformer.EntityRendererTransformer
  sereneseasons.asm.transformer.WorldTransformer
	GL info: ' Vendor: 'ATI Technologies Inc.' Version: '4.5.13491 Compatibility Profile Context 22.19.662.4' Renderer: 'AMD Radeon (TM) R9 390 Series'
	OptiFine Version: OptiFine_1.12.2_HD_U_D2
	OptiFine Build: 20180430-132249
	Render Distance Chunks: 16
	Mipmaps: 0
	Anisotropic Filtering: 1
	Antialiasing: 0
	Multitexture: false
	Shaders: (internal)
	OpenGlVersion: 4.5.13491 Compatibility Profile Context 22.19.662.4
	OpenGlRenderer: AMD Radeon (TM) R9 390 Series
	OpenGlVendor: ATI Technologies Inc.
	CpuCount: 8
```

</p>
</details>

**Comment by AlexModGuy** [comment]
fixed in 1.5.0
